tests/executables_src/testCircularDependencyFaultyFlags.cc

```diff
@@ -91,12 +130,78 @@ struct TestApplication1 : ctk::Application {
   ctk::ControlSystemModule cs;
 };
 
+template<typename APP_TYPE>
+struct CircularAppTestFixcture {
+  APP_TYPE app;
+  ctk::TestFacility test;
+
+  ctk::ScalarRegisterAccessor<int> a;
+  ctk::ScalarRegisterAccessor<int> b;
+  ctk::ScalarRegisterAccessor<int> C_trigger;
+  ctk::ScalarRegisterAccessor<int> A_out1;
+  ctk::ScalarRegisterAccessor<int> B_out1;
+  ctk::ScalarRegisterAccessor<int> C_out1;
+  ctk::ScalarRegisterAccessor<int> D_out1;
+  ctk::ScalarRegisterAccessor<int> A_in2;
+  ctk::ScalarRegisterAccessor<int> B_in2;
+  ctk::ScalarRegisterAccessor<int> C_in2;
+  ctk::ScalarRegisterAccessor<int> D_in2;
+  ctk::ScalarRegisterAccessor<int> circleResult;
+
+  void readAllLatest() {
+    A_out1.readLatest();
+    B_out1.readLatest();
+    C_out1.readLatest();
+    D_out1.readLatest();
+    A_in2.readLatest();
+    B_in2.readLatest();
+    C_in2.readLatest();
+    D_in2.readLatest();
+    circleResult.readLatest();
+  }
+
+  void checkAllDataValidity(ctk::DataValidity validity) {
+    BOOST_CHECK(A_out1.dataValidity() == validity);
+    BOOST_CHECK(B_out1.dataValidity() == validity);
+    BOOST_CHECK(C_out1.dataValidity() == validity);
+    BOOST_CHECK(D_out1.dataValidity() == validity);
+    BOOST_CHECK(A_in2.dataValidity() == validity);
+    BOOST_CHECK(B_in2.dataValidity() == validity);
+    BOOST_CHECK(C_in2.dataValidity() == validity);
+    BOOST_CHECK(D_in2.dataValidity() == validity);
+    BOOST_CHECK(circleResult.dataValidity() == validity);
+  }
+
+  CircularAppTestFixcture() {
+    test.runApplication();
+    a.replace(test.getScalar<int>("A/a"));
```

Comment:
Why don't you initialise the accessors via the brace initialiser? This can be done before runApplication() is called, you only have to instantiate the Application and the TestFacility first, but the order of instantiation is already like that.

Reply:
Wow, that saved many lines of unnecessary code. Thanks!